### PR TITLE
chore: replace RTB <Dialog>, <IconButton>

### PR DESCRIPTION
**Pull request by suprko**
#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/1333/desktop-replace-dialog-component
(also some leftover `Button` replacements; I hadn't replaced the `IconButtons`!)

#### Testing instructions  
Similar to https://github.com/PeerioTechnologies/peerio-desktop/pull/205 (but not nearly as big), this is a PR that affects a lot of little things in the app. All `IconButton`s have been replaced (they were missed on the previous PR), and all `Dialog`s as well.

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

**Comment by suprko** [comment]
@anri-asaturov looking for feedback on how I've set up `Dialog.jsx`. Specifically using `reaction` and `setTimeout` to make sure that the fade in/out transition plays. I tried to use `ref` for this, but I could only get that to work on component mount, not unmount.

**Comment by suprko** [comment]
^ and @seavan !

File: src/ui-library/Dialog.jsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        */
+        this.activeReaction = reaction(() => this.props.active, active => {
+            if (active) {
+                this.dialogRendered = true;
+                setTimeout(() => {
+                    this.dialogVisible = true;
+                }, 1);
+            } else {
+                this.dialogVisible = false;
+                setTimeout(() => {
```
Comment (anri-asaturov) [inline src/ui-library/Dialog.jsx]: 1. save return value of this `setTimeout` in this.unmountTimeout.
2. Inside the timeout function put this.unmountTimeout = null
2. after line 43 put `if(this.unmountTimeout) clearTimeout(this.unmountTimeout)`

If not sure why - ask me.

Comment (suprko) [reply src/ui-library/Dialog.jsx]: Pushed an update - like that?

What's the difference between doing it like this and using `clearTimeout`?

Comment (suprko) [reply src/ui-library/Dialog.jsx]: oh wait. misread your comment. editing :sweat_smile:

File: src/ui/AutoUpdateDialog.jsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ const React = require('react');
 const { observable } = require('mobx');
 const { observer } = require('mobx-react');
 const { t } = require('peerio-translator');
-const { Dialog } = require('~/react-toolbox');
+const { Dialog } = require('~/peer-ui');
```
Comment (anri-asaturov) [inline src/ui/AutoUpdateDialog.jsx]: You can delete this entire file

Comment (suprko) [reply src/ui/AutoUpdateDialog.jsx]: OK! There are a few files that are not being used but I wasn't sure which may come back and which will not be needed again.

Comment (anri-asaturov) [reply src/ui/AutoUpdateDialog.jsx]: We can recover files from git history, clean that stuff up :-D

Comment (anri-asaturov) [reply src/ui/AutoUpdateDialog.jsx]: Tell me when to merge

**Comment by suprko** [comment]
@anri-asaturov take another look at `Dialog.jsx`. Due to the way Dialogs are used in the app there are a lot of places where they are mounted but not rendered, so `keyup` subscription/unsub would not work properly in `component...Mount` functions and this was causing console errors. Had to move them into the `reaction` function.

If this looks OK then yes, ready to merge.

**Comment by anri-asaturov** [comment]
@suprko add  `window.removeEventListener('keyup', this.handleEscKey);` to conponentWillUnmount and we are safe

**Comment by anri-asaturov** [comment]
On another note - please refactor it a bit, make smaller functions out of the reaction so it's easier to read

**Comment by suprko** [comment]
@anri-asaturov `componentWillUnmount` doesn't work, some Dialogs are always mounted, just not rendered/`active`. e.g. the ContactProfile popup when you click on an avatar

**Comment by anri-asaturov** [comment]
@suprko I meant 'add another  window.removeEventListener('keyup', this.handleEscKey);' keep the one that's already in the reaction

**Comment by suprko** [comment]
@anri-asaturov Ah I see :+1:

**Comment by suprko** [comment]
@anri-asaturov done

**Comment by dchest** [comment]
I noticed both old and new <Dialog> sometimes allow changing focus on the content that it covers by pressing Tab. 

tabbing


If I remember correctly, this can be prevented by:

1. Setting focus on the dialog element when it appears.
2. Catching onBlur event, checking if the focus is outside the container, and if it is, putting it back inside the dialog.

**Comment by suprko** [comment]
@dchest good catch. I'll look into it

**Comment by dchest** [comment]
@suprko turns out native `<dialog>` is already in Electron, perhaps it would be easier to use it? I check and it handles focus nicely.

https://keithjgrant.com/posts/2018/meet-the-new-dialog-element/

Not sure if backdrop is stylable, but the dialog itself is:

screen shot 2018-01-17 at 17 27 21

**Comment by anri-asaturov** [comment]
great idea @dchest

**Pull request merged**